test/integration/AuthenticatedFileResourceStore.test.ts

```diff
@@ -0,0 +1,238 @@
+
+describe('A server using a FileResourceStore', (): void => {
+  let config: AuthenticatedFileResourceStoreConfig;
+  let handler: HttpHandler;
+  let store: ResourceStore;
+
+  beforeAll(
+    async(): Promise<void> => {
+      config = new AuthenticatedFileResourceStoreConfig();
+      handler = config.getHandler();
+      ({ store } = config);
+
+      await setAcl(
+        store,
+        'http://test.com/',
+        { read: true, write: true, append: true },
+        true,
+        true,
+        true,
+        undefined,
+        'agent',
+      );
+
+      // Check if permanent file exists, if not make one
+      let requestUrl = new URL('http://test.com/permanent.txt');
+      let response = await call(
+        handler,
+        requestUrl,
+        'GET',
+        { accept: 'text/*' },
+        [],
+      );
+      if (response.statusCode !== 200) {
+        requestUrl = new URL('http://test.com/');
+        const fileData = await fs.readFile('test/testfiles/permanent.txt');
+
+        response = await callFile(
+          handler,
+          requestUrl,
+          'POST',
+          {
+            'content-type': 'application/octet-stream',
+            slug: 'permanent.txt',
+            'transfer-encoding': 'chunked',
+          },
+          fileData,
+        );
+        expect(response.statusCode).toBe(200);
+      }
+    },
+  );
+  afterAll(
+    async(): Promise<void> => {
+      await setAcl(
+        store,
+        'http://test.com/',
+        { read: true, write: true, append: true },
+        true,
+        true,
+        true,
+        undefined,
+        'agent',
+      );
+
+      // Delete permanente file
+      const requestUrl = new URL('http://test.com/permanent.txt');
+      const response = await call(
+        handler,
+        requestUrl,
+        'DELETE',
+        {},
+        [],
+      );
+      expect(response.statusCode).toBe(200);
+
+      // Delete .acl file
+      await store.deleteResource({ path: 'http://test.com/.acl' });
+    },
+  );
+  describe('with acl', (): void => {
+    it('can add a file to the store, read it and delete it if allowed.', async(): Promise<
+    void
+    > => {
+      await setAcl(
+        store,
+        'http://test.com/',
+        { read: true, write: true, append: true },
+        true,
+        true,
+        true,
+        undefined,
+        'agent',
+      );
+
+      // POST
+      let requestUrl = new URL('http://test.com/');
+
+      const fileData = await fs.readFile('test/testfiles/testfile1.txt');
+
+      let response: MockResponse<any> = await callFile(
+        handler,
+        requestUrl,
+        'POST',
+        {
+          'content-type': 'application/octet-stream',
+          slug: 'testfile1.txt',
+          'transfer-encoding': 'chunked',
+        },
+        fileData,
+      );
+      expect(response.statusCode).toBe(200);
+      expect(response._getData()).toHaveLength(0);
+      const id = response._getHeaders().location;
```

Comment:
Is it okay to access underscore fields here? @joachimvh

Reply:
These are specific fields added by the mocking library, so yes.